File: app/src/main/java/uk/co/ourfriendirony/medianotifier/mediaitem/MediaItem.java
package uk.co.ourfriendirony.medianotifier.mediaitem;

import java.util.Date;
import java.util.List;

public interface MediaItem {
    String NO_DATE = "Date TBD";

    String getId();

    String getSubId();

    String getTitle();

    String getSubtitle();

    String getDescription();

    Date getReleaseDate();

    String getReleaseDateFull();

    String getReleaseDateYear();

    List<MediaItem> getChildren();

    void setChildren(List<MediaItem> children);

    int countChildren();

    String getExternalLink();

    Boolean getPlayed();
}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/mediaitem/MediaItem.kt
package uk.co.ourfriendirony.medianotifier.mediaitem

import java.util.*

interface MediaItem {
    val id: String
    val subId: String?
    val title: String?
    val subtitle: String?
    val description: String?
    val releaseDate: Date?
    val releaseDateFull: String
    val releaseDateYear: String
    var children: MutableList<MediaItem>
    val externalLink: String?
    val played: Boolean

    fun countChildren(): Int

    companion object {
        const val NO_DATE = "Date TBD"
    }
}